### PR TITLE
Dev cost management microsoft.cost management 2020 08 01 preview

### PR DESCRIPTION
**Pull request by vibsridh**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+  "paths": {
+  	"/{scope}/providers/Microsoft.CostManagement/insights": {
+      "get": {
+        "tags": [
+          "Insights"
+        ],
+        "operationId": "Insights_Get",
+        "description": "Get Insights for the given scope",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/costmanagement/"
```
[the remark this answers is not in the document]
Comment (flanakin) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json:44]: Not sure if we should be including language here

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+	  }
+  },
+  "definitions": {
+    "InsightsCollectionResponse": {
+      "description": "Result of listing Insights. It contains a list of available insight values.",
```
Comment (flanakin) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json:81]:
```suggestion
      "description": "Result of listing insights. It contains a list of available cost insights.",
```

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+          "title": "Insights properties"
+        }
+      }
+    },
+    "InsightsProperties": {
+      "type": "object",
+      "description": "The properties of the Insights.",
+      "properties": {
+        "status": {
+          "description": "Insight status kept from storage.",
```
Comment (flanakin) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json:120]: Can we include available values? I believe this should be defined/controlled by the API, right?

```suggestion
          "description": "Status of the insight.",
```

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+          "type": "string"
+        },
+        "eventDate": {
+          "description": "The date of event resulting in this anomaly.",
```
Comment (flanakin) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json:124]:
```suggestion
          "description": "The day the insight pertains to.",
```

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+          "type": "string",
+          "format": "date-time"
+        },
+        "scope": {
+          "description": "Cost Management scope for Insights in storage. This includes 'subscriptions/{subscriptionId}' for subscription scope",
```
Comment (flanakin) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json:129]:
```suggestion
          "description": "Resource Manager ID of the subscription, resource group, management group, or other billing scope this insight pertains to.",
```

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+          "type": "string",
+          "format": "date-time"
+        },
+        "timeDetected": {
+          "description": "The datetime when this anomaly was detected.",
+          "type": "string",
+          "format": "date-time"
+        },
+        "severity": {
+          "description": "Severity of anomaly to be used for surfacing High severity anomalies first.",
```
Comment (flanakin) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json:147]:
```suggestion
          "description": "Severity of the insight. Can be Critical, Major, High, Medium, Low, Minor, Awareness, Informational, or Verbose.",
```

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+          "type": "string"
+        },
+        "details": {
+          "description": "Details of anomaly.",
```
Comment (flanakin) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json:151]:
```suggestion
          "description": "Lower-level background information about this insight. Note that details schema changes based on kind.",
```

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+          "type": "string"
```
Comment (flanakin) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json:152]: I thought this was an object?

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+        },
+        "justification": {
+          "description": "Justification as why the impact qualifies as anomaly.",
```
Comment (flanakin) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json:155]:
```suggestion
          "description": "High-level summary information about this insight. Note that justification schema changes based on kind.",
```

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+          "type": "string"
```
Comment (flanakin) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json:156]: I thought this was an object?

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/examples/GetInsights.json [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+                "expectedValue": 53.69,
+                "deltaPercent": 0.1,
+                "expectedMin": 53.56,
+                "expectedMax": 53.89,
+                "isAnomaly": false,
+                "currency": "USD",
+                "average": 53.69,
+                "minimum": 48.88,
+                "maximum": 54.23,
+                "actual": 53.12
```
Comment (flanakin) [inline specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/examples/GetInsights.json:138]: Adding a new property that's coming soon:
```suggestion
                "actual": 53.12,
                "numberOfDays": 60
```

**Review by flanakin — CHANGES_REQUESTED** [review]

File: specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json (hunk 8 above)
Comment (vibsridh) [reply specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2020-08-01-preview/costmanagement.insights.json]: yes good catch! Changed

**Review by tadelesh — APPROVED** [review]

**Comment by leni-msft** [comment]
@vibsridh The "SDK azure-resource-manager-schemas" check failed due to Unable to determine scope for resource types scheduledActions . Though not blocking this PR, it affects ARM Template schemas autogeneration for your Swagger. Suggest to submit PR to update the config here: 
https://github.com/Azure/azure-resource-manager-schemas/blob/1836296c227f3651161e6af71852f389c4890bd7/generator/autogenlist.ts#L253-L271

**Pull request merged**